pyccel/ast/core.py

```diff
@@ -3149,90 +3149,135 @@ def __init__(
 
     @property
     def results(self):
+        """ List of variables which are the function results """
         return self._results
 
     @property
     def body(self):
+        """ CodeBlock containing all the statements in the function """
         return self._body
 
     @property
     def local_vars(self):
+        """ List of variables defined in the function """
         return self._local_vars
 
     @property
     def global_vars(self):
+        """ List of global variables used in the function """
         return self._global_vars
 
     @property
     def cls_name(self):
+        """ String containing the name of the class to which the method belongs.
+        If the function is not a class procedure then this returns None """
         return self._cls_name
 
     @property
     def hide(self):
+        """ False if function is pyccelized, True otherwise """
         return self._hide
 
     @property
     def kind(self):
+        """ Returns function or procedure """
         return self._kind
 
     @property
     def imports(self):
+        """ List of imports in the function """
         return self._imports
 
     @property
     def decorators(self):
+        """ List of decorators applied to the function """
         return self._decorators
 
     @property
     def headers(self):
+        """ List of headers applied to the function """
         return self._headers
 
     @property
     def templates(self):
+        """ List of templates used to determine the types """
         return self._templates
 
     @property
     def is_recursive(self):
+        """ Returns True if the function is recursive (i.e. calls itself)
+        and False otherwise """
         return self._is_recursive
 
     @property
     def is_pure(self):
+        """ Returns True if the function is marked as pure and False otherwise
+        Pure functions must not have any side effects.
+        In other words this means that the result must be the same no matter 
+        how many times the function is called
+        e.g:
+        >>> a = f()
+        >>> a = f()
+
+        gives the same result as
+        >>> a = f()
+
+        This is notably not true for I/O functions
+        """
         return self._is_pure
 
     @property
     def is_elemental(self):
+        """ returns True if the function is marked as elemental and
+        False otherwise
+        An elemental function is a function with a single scalar operator
+        and a scalar return value which can also be called on an array.
+        When it is called on an array it returns the result of the function
+        called elementwise on the array """
         return self._is_elemental
 
     @property
     def is_private(self):
+        """ True if the function should not be exposed to
+        other modules. This includes the wrapper module and
+        means that the function cannot be used in an import
+        or exposed to python """
         return self._is_private
```

Comment:
@saidctb @yguclu Does this method make sense when translating from python? It seems to add the `private :: ` suffix to the object in fortran. I remember having several discussions about this but I don't remember what the conclusion was

Reply:
probably it will be used when we introduce classes

Reply:
So then it is not used currently? That sounds to me like it can be removed for now?

Reply:
yes you can remove it

Reply:
Ah, it seems this is linked to a `@private` decorator. So maybe it is useful? @yguclu  do you have an opinion?

Reply:
This may be useful if a Python function can be translated to Fortran, but cannot be exposed back to Python through our wrapper... Do we actually process the `private` decorator at all?

Reply:
> This may be useful if a Python function can be translated to Fortran, but cannot be exposed back to Python through our wrapper

Maybe, but that isn't something that the user chooses. Currently we have functions as arguments which cannot be exposed to python. In the cwrapper these function prototypes are simply not printed and the wrapper just raises a NotImplementedError [here](https://github.com/pyccel/pyccel/blob/master/pyccel/codegen/printing/cwrappercode.py#L796)

>Do we actually process the private decorator at all?

It seems that we do. If I add a private decorator then the fortran code generated is correct, but pyccel then tries to pass it to the wrapper anyway

Reply:
OK. If we want to keep this decorator then we should make sure that it is not exposed to Python. If this amounts to adding an `if` clause in the right place, feel free to fix this in this PR. Please let me know what you find out.

Reply:
2 `if`s

Reply:
I have also added a very basic test